client/src/Components/PrivateRoute/PrivateRoute.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from "react";
+import { withRouter, Route, Redirect } from "react-router-dom";
+import { connect } from "react-redux";
+
+import HomeView from "../HomeView/HomeView";
+
+class PrivateRoute extends Component {
+  render() {
+    let protectedComponent = props => {
+      let { userRole, userID, component, tickets, path } = props;
+      if (userRole && userRole === "Owner") {
+        return <Route path={path} component={component} />;
+      } else if (userRole && userRole === "Worker") {
+        if (tickets.find(x => x.worker_id === userID)) {
+          return <Route path={path} component={HomeView} />;
```

Comment:
Is the home view the same place you're redirecting to in the else at the end?

Reply:
homeview is the list of tickets, the redirect at the end is to the login view